### DISCUSSION TITLE
Toggle and know when it changes to False/True

### DISCUSSION THREAD

**Discussion by 5675sp** [h2oai/wave#832] [Q&A] [answered]
As of now, I've created a ui.toggle name `toggle_enable_extend_all_waiting` therefore, I am listening to a change in its value as follows: 
```
if q.args.toggle_enable_extend_all_waiting:
        print('toggle has been click(value has change)')
```

But for some reason, when I click on the toggle, nothing is printed, and now I am wondering if this is a bug because q.args works with a ui.button; please advise; thank you.

**Comment by mtanco** [comment, marked as answer]
Hi @5675sp, the value of Toggles can be `True`, `False`, or `None` (visually False and None look the same in the UI). You if statement is only checking if the value of the toggle is True, so it won't catch the times that the trigger goes from False to True.

Please feel free to share a full minimal reproducible code snippet if you have more problems! 

As suggestion you might try `if q.args.toggle_enable_extend_all_waiting is not None`

**Reply by 5675sp** [reply]
I see but when the toggle is click I assume the value of it will change too True and the if statement I used will become true given the true value of toggle? Isn't that when a click is done the value is updated automatically?

**Reply by 5675sp** [reply]
With the `if q.args.toggle_enable_extend_all_waiting is not None` is still doesn't work. 

Here is the code I am using: 

```
    q.page['manage_active_lab_instances'] = ui.form_card(box='3 1 -1 -1', items=[
        ui.toggle(name='toggle_enable_extend_all_waiting', label='Enable Extend All Waiting')
           
    ])

    await q.page.save()

    if q.args.toggle_enable_extend_all_waiting is not None:
        print('TOGGLE CLICK')

```

**Reply by mtanco** [reply]
Oh I see - you need to add `trigger=True` to your toggle. Because it does not have a trigger that means it is not calling the server function when you change the toggle. This is just like having a dropdown list with no trigger where you can select items but it doesn't send the request to the server automatically. In your code we would expect there to be a button added for users to click and save the changes (or add `trigger=True`)

**Reply by mtanco** [reply]
Try this:

```
from h2o_wave import main, app, Q, ui, on, handle_on


@app('/')
async def serve(q: Q):

    if not q.client.init:
        # only add the element to the screen the first time, otherwise you will see "flashing" as the card is
        # repetitively added and the value of the item is receptively rewritten
        q.page['manage_active_lab_instances'] = ui.form_card(box='3 1 -1 -1', items=[
            ui.toggle(name='toggle_enable_extend_all_waiting', label='Enable Extend All Waiting', trigger=True)

        ])
        q.client.init = True

    if q.args.toggle_enable_extend_all_waiting is not None:
        print(f'TOGGLE CLICK: {q.args.toggle_enable_extend_all_waiting}')

        # trick app into thinking we changed the UI so that we don't see "loading"
        # remove this once we've actually added content to this clause
        q.page['dummy_for_testing'].items = []  

    await q.page.save()
```

**Reply by 5675sp** [reply]
Thank you @mtanco
